src/Internal/SetupDecryptionHandler.cs

```diff
@@ -201,6 +207,11 @@ protected async System.Threading.Tasks.Task DecryptObjectAsync(byte[] decryptedE
                         var instructionFileRequest = EncryptionUtils.GetInstructionFileRequest(getObjectResponse, EncryptionUtils.EncryptionInstructionFileSuffix);
                         instructionFileResponse = await GetInstructionFileAsync(instructionFileRequest).ConfigureAwait(false);
                     }
+                    catch (AmazonS3Exception amazonS3ExceptionInner) when (amazonS3ExceptionInner.ErrorCode == EncryptionUtils.NoSuchKey)
+                    {
+                        throw new AmazonServiceException($"Exception encountered while fetching Instruction File. Ensure the object you are" +
+                            $" attempting to decrypt has been encrypted using the S3 Encryption Client and instruction files are enabled.", amazonS3ExceptionInner);
```

Comment:
Can you remove the `and instructions files are enabled`. We are going in here as fallback after finding there is no encryption header metadata which is the more common approach. But there could be no encryption metadata for the same reason the object is not encrypted.